src/main/java/uk/gov/hmcts/darts/cases/service/impl/CaseServiceImpl.java

```diff
@@ -143,15 +145,15 @@ public PostCaseResponse addCaseOrUpdate(AddCaseRequest addCaseRequest) {
         CourtCaseEntity courtCase = retrieveCoreObjectService.retrieveOrCreateCase(
             addCaseRequest.getCourthouse(),
             addCaseRequest.getCaseNumber()
-        );
+        ).validateIsExpired();
```

Comment:
I'm not sure about adding this here because this is used by the "addCase" endpoint and we don't want to expose this case expired error to the external services which call that endpoint. 

I think we should ensure this validation is only enforced for endpoints which are used by the portal.

Reply:
Updated